clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3115,12 +3153,10 @@ static void handleWorkGroupSize(Sema &S, Decl *D, const ParsedAttr &AL) {
 
 // Handles work_group_size_hint.
 static void handleWorkGroupSizeHint(Sema &S, Decl *D, const ParsedAttr &AL) {
-  uint32_t WGSize[3];
-
-  if (!AL.checkExactlyNumArgs(S, 3))
```

Comment:
Do we not need this check for `WorkGroupSizeHint`

Reply:
Ok. I see there are tests checking this. Could you tell me where it is being enforced?

Reply:
This is handled automagically thanks to tablegen. The attribute lists three arguments, none of which are optional (https://github.com/intel/llvm/blob/sycl/clang/include/clang/Basic/Attr.td#L2968) and so the common handler gets it (https://github.com/intel/llvm/blob/sycl/clang/lib/Sema/SemaAttr.cpp#L1244).

Reply:
Thanks for explanation!